[Tablet Orders] Create order: Address design review items

iamgabrielma opened the pull request:
<!-- Remember about a good descriptive title. -->

Closes: #12155

## Description
This PR tackles the design review items logged for "Create Order", that is:

<img width="361" alt="Screenshot 2024-03-01 at 14 23 23" src="https://github.com/woocommerce/woocommerce-ios/assets/3812076/5c8d3834-2bdb-4f17-87a9-83c22d1eacce">

* Adds missing divider lines wrapping the `AddOrderComponentsSection` ("+ Add Shipping", "+ Add Coupon", ... )
* Makes all tappable buttons/rows 56pt height
* Reduces the existing space above and below button groups from 16pt to 8pt
* Adds a 8pt space to the top of the view

I have applied these without a conditional  `sideBySideViewForOrderForm` feature flag usage, since are design updates that are not coming only with this project but should affect the Order Form view across all its instances (split view, modal on modal, or plain iPhone). Happy to add additional logic if you think we should keep the previous UI if the flag is off.

## Testing instructions
1. With the feature flag off, observe that changes are applied to modal-on-modal, and iPhone
![simulator_screenshot_446BA2A5-074C-47BA-8476-A1A5ED79DC30](https://github.com/woocommerce/woocommerce-ios/assets/3812076/9e0267e8-1d1e-4d6a-b363-c6688ed2c0c1)

2. With the feature flag on, observe that changes are applied to side-by-side, and iPhone
![20240301-changes-applied-side-by-side](https://github.com/woocommerce/woocommerce-ios/assets/3812076/8813b98c-3c6d-4059-83e9-0a229852dc27)

Adding products, shipping, etc, ... should behave normally in all cases.

iamgabrielma committed:
Add dividers around AddOrderComponentsSection

iamgabrielma committed:
Reduce space between grouped views to 8pt (half)

iamgabrielma committed:
Add spacing section to the order’s view top

dangermattic commented:
<!--
  0 Errors
  0 Warnings
  1 Message: This PR is still a Draft: some...
  0 Markdowns
-->
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Message">
          1 Message
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:book:</td>
      <td data-sticky="false">This PR is still a Draft: some checks will be skipped.</td>
    </tr>
  </tbody>
</table>

<p align="right" data-meta="generated_by_pr-check">
  Generated by :no_entry_sign: <a href="https://danger.systems/">Danger</a>
</p>

iamgabrielma committed:
Make order component buttons 56p height

iamgabrielma committed:
Make order notes button 56p height

iamgabrielma committed:
Make customer notes button 56p height

iamgabrielma committed:
Make add product button 56p height

iamgabrielma committed:
Fix: Apply row height only when button is shown

The parent view is configured differently based on `shouldShowAddProductsButton`. If we set the frame of the button to be `rowHeight` directly without checking first if the row is rendered, it will be applied to the whole product section component, cutting off the products added to an order.

iamgabrielma committed:
Make custom amount button 56p height

iamgabrielma committed:
Remove redundant dividers

iamgabrielma commented:
👋 the failing test is unrelated to our changes, it comes from flakiness on `test_intervalGranularity_for_dates_with_days_difference_from_1_to_28()` which has already been fixed https://github.com/woocommerce/woocommerce-ios/pull/12157. Just waiting for it to be merged into trunk so we can update this branch to pass CI

iamgabrielma committed:
Merge branch 'trunk' into issue/12155-create-order-ui-updates

iamgabrielma committed:
Merge branch 'trunk' into issue/12155-create-order-ui-updates

joshheald committed:
12115 add missing divider in side-by-side mode

joshheald commented on WooCommerce/Classes/ViewRelated/Orders/Order Creation/AddOrderComponentsSection/AddOrderComponentsSection.swift:
```diff
@@ -45,10 +45,13 @@ struct AddOrderComponentsSection: View {
 
                 addGiftCardRow
+                    .frame(height: Constants.rowHeight)
             }
```
```suggestion
            Group {
                addShippingRow

                addCouponRow

                addGiftCardRow
            }
            .frame(minHeight: Constants.rowHeight)
```

joshheald replied:
Groups can be used to apply the same modifier to multiple views.

`minHeight` seems more appropriate to allow the buttons to grow with larger dynamic type.

iamgabrielma replied:
> Groups can be used to apply the same modifier to multiple views.

🤦 I knew this one, not sure why I didn't use it. Thanks for updating it!

> minHeight seems more appropriate to allow the buttons to grow with larger dynamic type.

That makes sense!

joshheald commented on WooCommerce/Classes/ViewRelated/Orders/Order Creation/CustomAmounts/OrderCustomAmountsSection.swift:
```diff
@@ -29,6 +29,7 @@ struct OrderCustomAmountsSection: View {
                 .accessibilityIdentifier(Accessibility.addCustomAmountIdentifier)
                 .buttonStyle(PlusButtonStyle())
             }
+            .frame(height: Layout.rowHeight)
```
```suggestion
            .frame(minHeight: Layout.rowHeight)
```

joshheald commented on WooCommerce/Classes/ViewRelated/Orders/Order Creation/CustomerNoteSection/CustomerNoteSection.swift:
```diff
@@ -73,12 +73,12 @@ private struct CustomerNoteSectionContent: View {
             }
             .buttonStyle(PlusButtonStyle())
-            .padding([.leading, .bottom, .trailing])
+            .frame(height: Layout.buttonHeight)
```
```suggestion
            .frame(minHeight: Layout.buttonHeight)
```

joshheald commented on WooCommerce/Classes/ViewRelated/Orders/Order Creation/CustomerSection/OrderCustomerSection.swift:
```diff
@@ -66,8 +66,8 @@ private struct OrderCustomerSectionContent: View {
             if !viewModel.isDataAvailable {
-                Spacer(minLength: Layout.verticalHeadlineSpacing)
                 createCustomerView
+                    .frame(height: Layout.buttonHeight)
```
```suggestion
                    .frame(minHeight: Layout.buttonHeight)
```

joshheald committed:
12155 Use minHeight to fix large dynamic type size

joshheald reviewed (approved):
Looks much better.

I added a divider that was missing at the top of the `Scan Products` line in side-by-side. Strange as it shows in your screenshot! 35b9100

Some suggestions to make the row height easier to apply and work better with dynamic type – I've applied them all in 01b83fc.

Ideally we would have this in some sort of style that got automatically applied to the rows, but I'm not sure how to do it off the top of my head, and it'll probably mean we don't fix/test other things well if we try

iamgabrielma commented:
> I added a divider that was missing at the top of the Scan Products line in side-by-side. Strange as it shows in your screenshot! https://github.com/woocommerce/woocommerce-ios/commit/35b910045ccdd667f442379fea13356220758da8

Good catch, I believe I know where this comes from: I realized we were applying a double `Divider()` to the top. The difference was not much, but enough to notice the 1px vs 2px in certain screens, so I removed it in the last commit here: https://github.com/woocommerce/woocommerce-ios/pull/12175/commits/94a8f008117a6d137c05957a109c4c6a400499c8 without realizing the second one was being applied to modal-on-modal only. Thanks for addressing it!
